cmd/crank/beta/top/top.go

```diff
@@ -0,0 +1,287 @@
+
+	ctx := context.Background()
+
+	pods, err := k8sClientset.CoreV1().Pods(c.Namespace).List(ctx, metav1.ListOptions{})
+
+	if err != nil {
+		return errors.Wrap(err, errFetchAllPods)
+	}
+
+	crossplanePods := getCrossplanePods(pods.Items)
+	logger.Debug("Fetched all Crossplane pods", "pods", crossplanePods, " from namespace", c.Namespace)
+
+	if len(crossplanePods) == 0 {
+		fmt.Println("No Crossplane pods found in the namespace", c.Namespace)
+		return nil
+	}
+
+	for i, pod := range crossplanePods {
+		podMetrics, err := metricsClientset.MetricsV1beta1().PodMetricses(pod.PodNamespace).Get(ctx, pod.PodName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, errAddingPodMetrics)
+		}
+		var totalCPUUsage, totalMemoryUsage int64
+		for _, container := range podMetrics.Containers {
+			totalCPUUsage += container.Usage.Cpu().ScaledValue(resource.Milli)
+			totalMemoryUsage += container.Usage.Memory().ScaledValue(resource.Mega)
+		}
+
+		crossplanePods[i].CPUUsage = totalCPUUsage
+		crossplanePods[i].MemoryUsage = fmt.Sprintf("%dMi", totalMemoryUsage)
+	}
+
+	if err != nil {
+		return errors.Wrap(err, errGetPodMetrics)
+	}
+	logger.Debug("Added metrics to Crossplane pods")
+
+	if c.Split {
+		printPodsByCategory(crossplanePods)
+		logger.Debug("Printed pods by category")
+	} else {
+		pods, err := printPodsTable(crossplanePods)
+		if err != nil {
+			return errors.Wrap(err, errCreatingPodsTable)
+		}
+		if _, err := fmt.Fprintln(tw, pods); err != nil {
+			return errors.Wrap(err, "error writing table content")
+		}
+		logger.Debug("Printed pods in a table")
+		return tw.Flush()
+	}
+	return nil
+}
+
+func printPodsTable(crossplanePods []topMetrics) (string, error) {
+	var builder strings.Builder
+
+	// Building header
+	headers := defaultPrinterRow{
+		podType:   "TYPE",
+		namespace: "NAMESPACE",
+		name:      "NAME",
+		cpu:       "CPU(milliCPU)",
+		memory:    "MEMORY(Mi)",
+	}
+	_, err := builder.WriteString(headers.String() + "\n")
+	if err != nil {
+		return "", errors.Wrap(err, errWriteHeader)
+	}
+
+	// Building rows for each pod
+	for _, pod := range crossplanePods {
+		row := defaultPrinterRow{
+			podType:   pod.PodType,
+			namespace: pod.PodNamespace,
+			name:      pod.PodName,
+			cpu:       fmt.Sprintf("%d", pod.CPUUsage),
+			memory:    pod.MemoryUsage,
+		}
+		_, err := builder.WriteString(row.String() + "\n")
+		if err != nil {
+			return "", errors.Wrap(err, errWriteRow)
+		}
+	}
+
+	return builder.String(), nil
+}
+
+func printPodsByCategory(pods []topMetrics) {
+	categorySummaries := make(map[string]struct {
+		Count       int
+		TotalCPU    int64
+		TotalMemory string
+	})
+
+	for _, pod := range pods {
+		summary := categorySummaries[pod.PodType]
+		summary.Count++
+		summary.TotalCPU += pod.CPUUsage
+
+		// Inline memory aggregation logic
+		if summary.TotalMemory == "" {
+			summary.TotalMemory = pod.MemoryUsage
+		} else {
+			// Concatenate for simplicity; replace with actual aggregation logic
```

Comment:
is this a TODO?

Reply:
It is now :). Another leftover from the summary implementation attempt.